src/clusterfuzz/_internal/build_management/build_manager.py

```diff
@@ -1199,8 +1202,64 @@ def _get_latest_revision(bucket_paths):
   return None
 
 
-def setup_trunk_build(bucket_paths, fuzz_target, build_prefix=None):
+def _emit_build_age_metric(gcs_path):
+  """Emits a metric to track the age of a build."""
+  try:
+    last_update_time = storage.get(gcs_path).get('updated')
+    # TODO(vitorguidi): standardize return type between fs and gcs.
+    if isinstance(last_update_time, str):
+      # storage.get returns two different types for the updated field:
+      # the gcs api returns string, and the local filesystem implementation
+      # returns a datetime.datetime object normalized for UTC.
+      last_update_time = datetime.datetime.fromisoformat(last_update_time)
+    now = datetime.datetime.now(datetime.timezone.utc)
+    elapsed_time = now - last_update_time
+    elapsed_time_in_hours = elapsed_time.total_seconds() / 3600
+    # Fuzz targets do not apply for custom builds
+    labels = {
+        'job': os.getenv('JOB_NAME'),
+        'platform': environment.platform(),
+        'task': os.getenv('TASK_NAME'),
+    }
+    monitoring_metrics.JOB_BUILD_AGE.add(elapsed_time_in_hours, labels)
+    # This field is expected as a datetime object
+    # https://cloud.google.com/storage/docs/json_api/v1/objects#resource
+  except Exception as e:
+    logs.error(f'Failed to emit build age metric for {gcs_path}: {e}')
+
+
+def _get_build_url(bucket_path: Optional[str], revision: int,
+                   job_type: Optional[str]):
+  """Returns the GCS url for a build, given a bucket path and revision"""
+  build_urls = get_build_urls_list(bucket_path)
+  if not build_urls:
+    logs.error('Error getting build urls for job %s.' % job_type)
+    return None
+  build_url = revisions.find_build_url(bucket_path, build_urls, revision)
+  if not build_url:
+    logs.error(
+        'Error getting build url for job %s (r%d).' % (job_type, revision))
+    return None
+  return build_url
+
+
+def _get_build_bucket_paths():
+  bucket_paths = []
+  for env_var in DEFAULT_BUILD_BUCKET_PATH_ENV_VARS:
+    bucket_path = get_bucket_path(env_var)
+    if bucket_path:
+      bucket_paths.append(bucket_path)
+    else:
+      logs.info('Bucket path not found for %s' % env_var)
+  return bucket_paths
+
+
+def setup_trunk_build(fuzz_target, build_prefix=None):
   """Sets up latest trunk build."""
+  bucket_paths = _get_build_bucket_paths()
+  if len(bucket_paths) == 0:
```

Comment:
nit: just `if not bucket_paths` per https://google.github.io/styleguide/pyguide.html#214-truefalse-evaluations